lib/matplotlib/axes/_base.py

```diff
@@ -4344,7 +4317,10 @@ def _set_view_from_bbox(self, bbox, direction='in',
             x_trf = self.xaxis.get_transform()
             sxmin0, sxmax0, sxmin, sxmax = x_trf.transform(
                 [xmin0, xmax0, xmin, xmax])  # To screen space.
-            factor = (sxmax0 - sxmin0) / (sxmax - sxmin)  # Unzoom factor.
+            with np.errstate(divide='ignore'):
+                # divide by zero is possible for colorbars where the zoom
+                # isn't canceled for small values
+                factor = (sxmax0 - sxmin0) / (sxmax - sxmin)  # Unzoom factor.
```

Comment:
So you're supposed to end up with `factor = inf` if `sxmax == sxmin`?  Then `sxmin1 = -inf; sxmax1  = +inf`?  Can you clarify the cases where this happens?  (Ditto below.)

Reply:
Your other comments about forcing the "key" for the zoom actually fixed this here, so I've removed it now.

For the curious: It was happening because the key modifier wasn't pushing sxmax/sxmin out to the edges of the short axis, so the short-axis factor could be non-sensical, but was never used so I was ignoring it.